airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/write/object_storage/FilePartAccumulator.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.write.object_storage
+
+import com.google.common.collect.Range
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings
+import io.airbyte.cdk.load.command.DestinationStream
+import io.airbyte.cdk.load.command.object_storage.ObjectStorageUploadConfiguration
+import io.airbyte.cdk.load.file.object_storage.ObjectStoragePathFactory
+import io.airbyte.cdk.load.file.object_storage.PartFactory
+import io.airbyte.cdk.load.message.Batch
+import io.airbyte.cdk.load.message.BatchEnvelope
+import io.airbyte.cdk.load.message.DestinationFile
+import io.airbyte.cdk.load.message.MultiProducerChannel
+import io.airbyte.cdk.load.message.object_storage.LoadablePart
+import io.airbyte.cdk.load.write.BatchAccumulator
+import java.io.File
+import java.nio.file.Path
+
+@SuppressFBWarnings(
+    "NP_NONNULL_PARAM_VIOLATION",
+    justification = "state is guaranteed to be non-null by Kotlin's type system"
+)
+class FilePartAccumulator(
```

Comment:
Can we add a comment here flagging that this is nonstandard usage. Specifically, the accumulators shouldn't be publishing to queues or doing bookkeeping, it's just awkward to do the file work without more of a flow interface. (because it goes `1 Record -> Many Parts` instead of `Many Records >- 1 Part`)

Like really, this should be `FilePartGenerator` (from `StreamLoader::createFilePartGenerator`) with something like

```
class FilePartGenerator(
  file: DestinationFile
)  {
  fun parts(): Sequence<Part> { ... }
}
```

Reply:
(But no change required, since you've already tested this. I just don't want people to follow this pattern for non-file stuff.)

Reply:
I added a comment explaining why we need this pattern